dbt/adapters/bigquery/connections.py

```diff
@@ -713,8 +714,24 @@ def _query_and_results(
                 self._bq_job_link(query_job.location, query_job.project, query_job.job_id)
             )
 
-        iterator = query_job.result(max_results=limit, timeout=job_execution_timeout)
+        # only use async logic if user specifies a timeout
+        if job_execution_timeout:
```

Comment:
This could be overkill, but I could see an argument for putting all of this new code into a private method so that the code here becomes something like:
```python
if job_execution_timeout:
    iterator = self._run_timeout_query(query_job, timeout, limit)
else:
    iterator = query_job.result(max_results=limit)
return query_job, iterator
```

Reply:
I went back and forth on that, what I'd really like to do is abstract all of this out of the class to separate function. Agreed that this feels like a it's own method but I'm not sure if a private method would really help or just add more clutter to an already overloaded class